src/oop/special_class_method/medium/q5/MainTest.java

```diff
@@ -0,0 +1,174 @@
+                {5.67, true},
+                {Math.PI, false}
+        };
+        for (int i = 0; i < tests.length; i++) {
+            double value = (double) tests[i][0];
+            boolean on = (boolean) tests[i][1];
+            Object[][] arguments = {
+                    {value, double.class},
+                    {on, boolean.class}
+            };
+            Class<?>[] classes = {double.class, boolean.class};
+            Object testInstance = testClass.createInstance(arguments);
+            _assertEquals(value * 7, testClass.getFieldValue(testInstance, attributeName1), modifiedClassMessage);
+            _assertEquals(on, testClass.getFieldValue(testInstance, attributeName2), modifiedClassMessage);
+            assertTrue(testClass.hasModifier(classes, "public"), modifiedClassMessage);
+        }
+    }
+
+    private static Stream<Arguments> booleanDoublesInputProvider() {
+        return Stream.of(
+                Arguments.of(4.23232, false),
+                Arguments.of(343434.44364, true),
+                Arguments.of(5.67, true),
+                Arguments.of(Math.PI, false)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("booleanDoublesInputProvider")
+    public void correctGetSumMethod(double sum, boolean correct) throws Throwable {
+        Object[][] arguments = {
+                {sum, double.class},
+                {correct, boolean.class}
+        };
+        Object testInstance = testClass.createInstance(arguments);
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName1, testInstance);
+        assertTrue(testClass.hasModifier(getAttributeMethodName1, null, "public"),
+                "Your " + getAttributeMethodName1 + " method does not have the correct visibility modifier.");
+        _assertEquals(sum * 7, getMethodOutput,
+                "Your " + getAttributeMethodName1 + " method does not return the value of the size attribute.");
+    }
+
+    private static Stream<Arguments> twoDoubleInputProvider() {
+        return Stream.of(
+                Arguments.of(5.4, true, 6.4),
+                Arguments.of(654.3, false, 54.2),
+                Arguments.of(43.3, true, 43.2),
+                Arguments.of(432.2, false, 899.8)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("twoDoubleInputProvider")
+    public void correctSetSumMethod(double initialValue, boolean c, double updatedValue) throws Throwable {
+        Object[][] instantiationArguments = {
+                {initialValue, double.class},
+                {c, boolean.class}
+        };
+        Object testInstance = testClass.createInstance(instantiationArguments);
+        Object[][] setMethodArguments = {
+                {updatedValue, double.class}
+        };
+        Class<?>[] methodModifierClasses = {
+                double.class
+        };
+        Object setMethodOutput = testClass.callMethod(setAttributeMethodName1, setMethodArguments, testInstance);
```

Comment:
We should probably have two separate tests for the method name and the arguments for better error messages